data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/events/EventSubscriber.java

```diff
@@ -124,18 +126,40 @@ private void processEventQueue() {
                     new ByteArrayInputStream(event.getMessageData().toArrayUnsafe()));
                 event = queuedEvents.poll();
               }
+              successiveFailureCounter = 0;
             })
         .alwaysRun(
             () -> {
               processingQueue.set(false);
-              if (queuedEvents.size() > 0) {
+              if (!queuedEvents.isEmpty()) {
                 processEventQueue();
               }
             })
         .finish(
-            error ->
-                LOG.error(
-                    "Failed to process event queue for client " + sseClient.hashCode(), error));
+            error -> {
+              successiveFailureCounter++;
```

Comment:
I suspect there might be some concurrency problems here.
the previous `.alwasyRun` opens the floodgate again before `.finish` runs, potentially resetting the counter to 0?

Reply:
that'd be fine, if it resets to 0 we can continue and not nuke the connection... but if it goes over that limit we should really give up and not keep trying to send on blocked ports continuously...

Reply:
k so I'd suggest to add `volatile` to it, just to make sure threads always read the last updated value.